### PR TITLE
MoveAlongPath for docs

### PR DESCRIPTION
**Pull request by kolibril13**
## Motivation

Some improvements for the docs
## Oneline Summary of Changes

```
- Added MoveAlongPath example to the docs (:pr:`PR NUMBER HERE`)
```
## Reviewer Checklist
- [x] Newly added functions/classes are either private or have a docstring
- [x] Newly added functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [x] Newly added documentation builds, looks correctly formatted, and adds no additional build warnings
- [ ] The oneline summary has been included [in the wiki](https://github.com/ManimCommunity/manim/wiki/Changelog-for-next-release)

### CODE REVIEW

**Comment by kolibril13** [comment]
There are also some niche animations like PhaseFlow and ComplexHomotopy, SmoothedVectorizedHomotopy.
What do you think about depreciating them?
Or if not, should we add also examples for them?
We could also make an "Animations" overview like the one here:
https://docs.manim.community/en/latest/reference/manim.animation.creation.html?highlight=creation#module-manim.animation.creation

File: manim/animation/movement.py [hunk 1]
```diff
@@ -110,6 +110,21 @@ def interpolate_mobject(self, alpha: float) -> None:
...
+    .. manim:: MoveAlongPathExample
+
+        class MoveAlongPathExample(Scene):
+            def construct(self):
+                d1= Dot()
+                l1 = Line(LEFT,RIGHT)
+                l2= VMobject()
+                self.add(d1,l1,l2)
+                l2.add_updater(lambda x: x.become(Line(LEFT, d1.get_center()).set_color(ORANGE)))
+                self.play(MoveAlongPath(d1,l1), rate_func= linear)
```
Comment (behackl) [inline manim/animation/movement.py:125]: Just some inconsistencies in the spacing.

```suggestion
                d1 = Dot()
                l1 = Line(LEFT, RIGHT)
                l2 = VMobject()
                self.add(d1, l1, l2)
                l2.add_updater(lambda x: x.become(Line(LEFT, d1.get_center()).set_color(ORANGE)))
                self.play(MoveAlongPath(d1, l1), rate_func=linear)
```

File: manim/animation/movement.py [hunk 2]
```diff
@@ -110,6 +110,21 @@ def interpolate_mobject(self, alpha: float) -> None:
...
+    --------
+    .. manim:: MoveAlongPathExample
+
+        class MoveAlongPathExample(Scene):
+            def construct(self):
+                d1= Dot()
+                l1 = Line(LEFT,RIGHT)
+                l2= VMobject()
+                self.add(d1,l1,l2)
+                l2.add_updater(lambda x: x.become(Line(LEFT, d1.get_center()).set_color(ORANGE)))
```
Comment (behackl) [inline manim/animation/movement.py:124]: Comment, non-blocking: I think the animation would be nicer if either the moving Dot would also be orange, or if the orange line would be below the dot (hello, `z_index` again!). :-)

**Review by behackl — APPROVED** [review]
In general, these changes LGTM; left two minor comments.

File: manim/animation/movement.py [hunk 3]
```diff
@@ -110,6 +110,21 @@ def interpolate_mobject(self, alpha: float) -> None:
 
 
 class MoveAlongPath(Animation):
+    """Make Mobject m1 move along path of Mobject m2.
```
Comment (leotrs) [inline manim/animation/movement.py:113]: Comment: usually you want class/method docstrings to _not_ mention parameters by name.

```suggestion
    """Make one mobject move along the path of another mobject.
```

File: manim/animation/movement.py [hunk 4]
```diff
@@ -110,6 +110,21 @@ def interpolate_mobject(self, alpha: float) -> None:
...
 
 class MoveAlongPath(Animation):
+    """Make one mobject move along the path of another mobject.
+    Example
+    --------
+    .. manim:: MoveAlongPathExample
+
+        class MoveAlongPathExample(Scene):
+            def construct(self):
+                d1 = Dot()
```
Comment (kolibril13) [inline manim/animation/movement.py:120]:
```suggestion
                d1 = Dot().set_color(ORANGE)
```

File: manim/animation/movement.py (hunk 2 above)
Comment (kolibril13) [reply manim/animation/movement.py:124]: done ;)

**Review by leotrs — APPROVED** [review]

**Pull request merged**